docs(k8s): add monitoring docs (#1966)
scaleway/docs-content: containers/kubernetes/how-to/monitor-cluster.mdx

---
meta:
  title: How to monitor your Kubernetes Kapsule cluster with Cockpit
  description: This page explains how to monitor your Kubernetes Kapsule cluster with Cockpit
content:
  h1: How to monitor your Kubernetes Kapsule cluster with Cockpit
  paragraph: This page explains how to monitor your Kubernetes Kapsule cluster with Cockpit
tags: kubernetes kubernetes-kapsule kapsule cluster observability cockpit
dates:

>   validation: 2023-07-24

Author: ```suggestion
  validation: 2023-08-08
```

  posted: 2023-07-24
categories:
  - kubernetes
---

<Macro id="iam-requirements" />

<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
  - You have [created](/containers/kubernetes/how-to/create-cluster) a Kubernetes Kapsule cluster
  - You have [activated](/observability/cockpit/how-to/activate-cockpit/) your observability Cockpit
  - You have [created](/observability/cockpit/how-to/retrieve-grafana-credentials/) a Grafana user for Cockpit
</Message>

> Scaleway's observability Cockpit provides automated monitoring of your Kubernetes cluster. As Kubernetes environments grow in complexity, monitoring becomes crucial for optimal performance. Visualization of the clusters' metrics are available in a hosted Grafana.

Reviewer: ```suggestion
Scaleway's observability Cockpit provides automated monitoring of your Kubernetes cluster. As Kubernetes environments grow in complexity, monitoring becomes crucial for optimal performance. Visualization of the clusters' metrics is available in a hosted Grafana.
```

    <Lightbox src="scaleway-k8s-cluster-metrics.webp" alt="" />
    <Message type="tip">
      Select the cluster you want to display from the drop-down list at the top of the page to change the currently displayed cluster.
    </Message>

### Monitoring of the kube-apiserver

The cluster overview dashboard offers real-time monitoring capabilities for the `kube-apiserver` within your Kubernetes cluster. Serving as a crucial component on the control plane, the `kube-apiserver` acts as the gateway to the Kubernetes API, allowing you to interact with the cluster.

Large clusters with numerous resources (nodes, pods, and custom resources (CRDs)) and high controller requests (e.g. argocd and velero) can cause CPU and memory spikes, leading to sluggish or unresponsive API server performance. You may also encounter errors like `"EOF"` when using `kubectl`.

To address this potential issue, it is crucial to monitor the CPU and RAM consumption of the apiserver closely. By doing so, you can proactively manage and reduce the load on the apiserver, thus averting performance bottlenecks.

> For scenarios where you require consistent and predictable performance, along with dedicated resources for the apiserver, consider migrating it to a [dedicated contol plane](https://www.scaleway.com/en/kubernetes-dedicated-control-plane/) to ensure optimal performance and stability.

Reviewer 2: ```suggestion
For scenarios where you require consistent and predictable performance, along with dedicated resources for the apiserver, consider migrating it to a [dedicated control plane](https://www.scaleway.com/en/kubernetes-dedicated-control-plane/) to ensure optimal performance and stability.
```

## How to view the Kubernetes control plane logs

1. Open your Cockpit in the [Scaleway console](https://console.scaleway.com/cockpit/overview).
2. Click **Manage Scaleway data**. The Grafana interface opens in a new browser tab.
3. Log into Grafana using your Cockpits' Grafana user and password. The list of your Cockpit dashboards displays.
4. Click **Kubernetes Cluster Logs** in the list of dashboards. A listing of your control plane logs displays. This dashboard allows you to view the logs of multiple components of your Kubernetes cluster, such as the `controller-manager`, `ccm`, `kapsule-autoscaler`, and more.
    <Message type="note">

>       Currently no logs from the `kube-apiserver` are sent to your Cockpit.

Reviewer 2: ```suggestion
      Currently, no logs from the `kube-apiserver` are sent to your Cockpit.
```

[Cockpit](/observability/cockpit/quickstart/) empowers you to monitor the clusters' control plane, nodes, managed resources and the cluster system applications.
* Detect incidents and diagnose failures in the earliest stages
* Get new insights about control plane and worker nodes usage such as CPU, memory, to optimize your resource allocation
* Achieve a better understanding of your infrastructure by analyzing data on your API server, system applications, up and down scaling behavior

## How to access your Cockpit Grafana dashboard

<Message type="important">
  You must have [activated](/observability/cockpit/how-to/activate-cockpit/) your Cockpit. Cockpit’s free plan retention time being 31 days for metrics and 7 days for logs.
</Message>

1. Open your Cockpit in the [Scaleway console](https://console.scaleway.com/cockpit/overview).
2. Click **Manage Scaleway data**. The Grafana interface opens in a new browser tab.
3. Log into Grafana using your Cockpits' Grafana user and password. The list of your Cockpit dashboards displays.

> 4. Click **Kubernetes Cluster Overview** in the list of dashboards. A status overview of your cluster displays. This dashboard allows you to monitor multiple components of your Kubernetes cluster: control-plane, nodes, managed resources and the cluster system applications.

Reviewer 3: ```suggestion
4. Click **Kubernetes Cluster Overview** in the list of dashboards. A status overview of your cluster displays. This dashboard allows you to monitor multiple components of your Kubernetes cluster: control-plane, nodes, managed resources and cluster system applications.
```

> 4. Click **Kubernetes Cluster Overview** in the list of dashboards. A status overview of your cluster displays. This dashboard allows you to monitor multiple components of your Kubernetes cluster: control-plane, nodes, managed resources and the cluster system applications.

Reviewer 3: ```suggestion
4. Click **Kubernetes Cluster Overview** in the list of dashboards. A status overview of your cluster displays. This dashboard allows you to monitor multiple components of your Kubernetes cluster: control-plane, nodes, managed resources and cluster system applications.
```

<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com) 
  - You have [created](/containers/kubernetes/how-to/create-cluster) a Kubernetes Kapsule cluster
  - You have [activated](/observability/cockpit/how-to/activate-cockpit/) your observability Cockpit
</Message>

Scaleway's observability Cockpit provides automated monitoring of your Kubernetes cluster. As Kubernetes environments grow in complexity, monitoring becomes crucial for optimal performance. Visualization of the clusters' metrics are available in a hosted Grafana.
Grafana's rich visualizations and ease of use make it an ideal choice. Cockpit offers a hassle-free solution, providing real-time insights without extra infrastructure.

[Cockpit](/observability/cockpit/quickstart/) empowers you to monitor the clusters' control plane, nodes, managed resources and the cluster system applications.
* Detect incidents and diagnose failures in the earliest stages

> * Get new insights about control plane and worker nodes usage such as CPU, memory, to optimize your resource allocation

Reviewer 4: ```suggestion
* Get new insights on control plane and worker nodes usage such as CPU and memory to optimize your resource allocation
```

* Achieve a better understanding of your infrastructure by analyzing data on your API server, system applications, up and down scaling behavior

## How to access your Cockpit Grafana dashboard

<Message type="important">

>   You must have [activated](/observability/cockpit/how-to/activate-cockpit/) your Cockpit. Cockpit’s free plan retention time being 31 days for metrics and 7 days for logs.

Reviewer 4: ```suggestion
  You must have [activated](/observability/cockpit/how-to/activate-cockpit/) your Cockpit. Cockpit’s free plan's retention time is 31 days for metrics and 7 days for logs.
```

</Message>

> 1. Open your Cockpit in the [Scaleway console](https://console.scaleway.com/cockpit/overview).

Reviewer 4: There should be a step after this one to create a Grafana user, so they are able to log in to Grafana with their credentials or maybe add it to the requirements

2. Click **Manage Scaleway data**. The Grafana interface opens in a new browser tab.
3. Log into Grafana using your Cockpits' Grafana user and password. The list of your Cockpit dashboards displays.
4. Click **Kubernetes Cluster Overview** in the list of dashboards. A status overview of your cluster displays. This dashboard allows you to monitor multiple components of your Kubernetes cluster: control-plane, nodes, managed resources and cluster system applications.
    <Lightbox src="scaleway-k8s-cluster-metrics.webp" alt="" />
    <Message type="tip">
      Select the cluster you want to display from the drop-down list at the top of the page to change the currently displayed cluster.
    </Message>

### Monitoring of the kube-apiserver

The cluster overview dashboard offers real-time monitoring capabilities for the `kube-apiserver` within your Kubernetes cluster. Serving as a crucial component on the control plane, the `kube-apiserver` acts as the gateway to the Kubernetes API, allowing you to interact with the cluster.

> In cases where your clusters are large and containing numerous resources such as nodes, pods, and custom resources (CRDs), or when experiencing a large number of simultaneous requests to the apiserver from controllers like argocd and velero,  it is possible for the CPU and memory usage to spike. Consequently, the performance of the apiserver may noticeably decline, leading to sluggishness or unresponsiveness. You might encounter errors like `"EOF"` when using `kubectl`.

Reviewer 4: This sentence is a bit long, maybe try making it shorter?

Author: Can't make it shorter. If you have any idea how, let me know

Reviewer 4: The CPU and memory usage might spike for large clusters containing many resources (nodes, pods, and custom resources) or experiencing many simultaneous requests to the API server from controllers such as argocd and velero. 

Something like that maybe ?

> In cases where your clusters are large and containing numerous resources such as nodes, pods, and custom resources (CRDs), or when experiencing a large number of simultaneous requests to the apiserver from controllers like argocd and velero,  it is possible for the CPU and memory usage to spike. Consequently, the performance of the apiserver may noticeably decline, leading to sluggishness or unresponsiveness. You might encounter errors like `"EOF"` when using `kubectl`.

Reviewer 4: ```suggestion
In cases where your clusters are large and contain many resources (nodes, pods, and custom resources (CRDs)), or when experiencing a large number of simultaneous requests to the apiserver from controllers (such as  argocd and velero),  it is possible for the CPU and memory usage to spike. Consequently, the performance of the apiserver may noticeably decline, leading to sluggishness or unresponsiveness. You might encounter errors like `"EOF"` when using `kubectl`.
```
